### PR TITLE
Access control for submission pages

### PR DESCRIPTION
**Pull request by hlminh2000**

### CODE REVIEW

File: client/components/pages/submission-system/NavBar.js [hunk 1]
```diff
@@ -57,6 +57,9 @@ export default (props: { path?: string, logOut: void => void, children?: React.N
     }
   })();
 
+  const canAccessSubmission =
+    !!egoJwt && (canReadProgram({ egoJwt, programId: '' }) || isRdpcMember(egoJwt));
```
Comment (joneubank) [inline client/components/pages/submission-system/NavBar.js]: Is this correct? how do you check permission without the programId?

Comment (joneubank) [reply client/components/pages/submission-system/NavBar.js]: If this is the syntax for ability to read ALL programs... we should maybe make an explicit method for that check

File: client/components/pages/submission-system/programs/index.js [hunk 2]
```diff
@@ -85,9 +86,11 @@ export default ({
           >
             All Programs
           </Typography>
-          <Link href={CREATE_PROGRAM_PAGE_PATH}>
-            <Button>Create a program</Button>
-          </Link>
+          {isDccMember(egoJwt) && (
+            <Link href={CREATE_PROGRAM_PAGE_PATH}>
```
Comment (joneubank) [inline client/components/pages/submission-system/programs/index.js]: 👍

File: client/package.json [hunk 3]
```diff
@@ -27,7 +27,7 @@
     "@emotion/babel-preset-css-prop": "^10.0.9",
     "@emotion/core": "^10.0.10",
     "@emotion/styled": "^10.0.11",
-    "@icgc-argo/ego-token-utils": "^3.0.0",
+    "@icgc-argo/ego-token-utils": "^3.1.0",
```
Comment (joneubank) [inline client/package.json]: We're at version 3 already?!

File: client/components/pages/submission-system/NavBar.js (hunk 1 above)
Comment (hlminh2000) [reply client/components/pages/submission-system/NavBar.js]: lol yes :))). Alternatiely, maybe default `programId` to `""`, then run `canReadProgram({ egoJwt })`? it's basically ability to read any program at all :P

File: client/package.json (hunk 3 above)
Comment (hlminh2000) [reply client/package.json]: 1 and 2 were my practice releases that failed lol

File: client/components/pages/submission-system/NavBar.js (hunk 1 above)
Comment (joneubank) [reply client/components/pages/submission-system/NavBar.js]: Yeah lets make it that for the future, and document the special case. What I don't want is a partial string match against programID being done inside the ego token utils. Should be exact match against program or empty string check for any program.

Comment (joneubank) [reply client/components/pages/submission-system/NavBar.js]: on second though, after our outside discussion... lets have a separate method for this check, not a special case of canReadProgram()

File: client/pages/submission/program/index.js [hunk 4]
```diff
@@ -10,6 +10,7 @@ import { programsListQuery } from 'components/pages/submission-system/programs/q
 
 export default createPage({
   isPublic: false,
-  isAccessible: async ({ egoJwt, ctx }) => !isRdpcMember(egoJwt),
+  isAccessible: async ({ egoJwt, ctx }) =>
+    !isRdpcMember(egoJwt) && canReadProgram({ egoJwt, programId: '' }),
```
Comment (joneubank) [inline client/pages/submission/program/index.js]: Can you clarify the rule here for me? RDPC users can't access program list? What about DCCMember (not included in this access rule)

Comment (hlminh2000) [reply client/pages/submission/program/index.js]: `canReadProgram` already handles `dcc` users

Comment (hlminh2000) [reply client/pages/submission/program/index.js]: and yes, `RDPC` can't access program list, as per https://wiki.oicr.on.ca/display/icgcargotech/Sitemap#Sitemap-PermissionsMatrixforSiteMap

Comment (joneubank) [reply client/pages/submission/program/index.js]: thanks. the ego token utils changes we discussed will make this clearer for future generations. 👩‍🚀

**Pull request merged**